tests/system/providers/google/cloud/sql_to_sheets/example_sql_to_sheets.py

```diff
@@ -86,16 +134,152 @@
     catchup=False,
     tags=["example", "sql"],
 ) as dag:
+    create_cloud_sql_instance = CloudSQLCreateInstanceOperator(
+        task_id="create_cloud_sql_instance",
+        project_id=PROJECT_ID,
+        instance=CLOUD_SQL_INSTANCE,
+        body=CLOUD_SQL_INSTANCE_CREATION_BODY,
+    )
+
+    create_database = CloudSQLCreateInstanceDatabaseOperator(
+        task_id="create_database", body=DB_CREATE_BODY, instance=CLOUD_SQL_INSTANCE
+    )
+
+    @task
+    def create_user() -> None:
+        with discovery.build("sqladmin", "v1beta4") as service:
+            request = service.users().insert(
+                project=PROJECT_ID,
+                instance=CLOUD_SQL_INSTANCE,
+                body={
+                    "name": DB_USER_NAME,
+                    "password": DB_USER_PASSWORD,
+                },
+            )
+            request.execute()
+
+    create_user_task = create_user()
+
+    @task
+    def get_public_ip() -> str | None:
+        with discovery.build("sqladmin", "v1beta4") as service:
+            request = service.connect().get(
+                project=PROJECT_ID, instance=CLOUD_SQL_INSTANCE, fields="ipAddresses"
+            )
+            response = request.execute()
+            for ip_item in response.get("ipAddresses", []):
+                if ip_item["type"] == "PRIMARY":
+                    return ip_item["ipAddress"]
+
+    get_public_ip_task = get_public_ip()
+
+    @task
+    def setup_postgres_connection(**kwargs) -> None:
+        public_ip = kwargs["ti"].xcom_pull(task_ids="get_public_ip")
+        connection = Connection(
+            conn_id=DB_CONNECTION_ID,
+            description="Example PostgreSQL connection",
+            conn_type="postgres",
+            host=public_ip,
+            login=DB_USER_NAME,
+            password=DB_USER_PASSWORD,
+            schema=DB_NAME,
+            port=DB_PORT,
+        )
+        session: Session = Session()
+        if session.query(Connection).filter(Connection.conn_id == DB_CONNECTION_ID).first():
+            log.warning("Connection %s already exists", DB_CONNECTION_ID)
+            return None
+
+        session.add(connection)
+        session.commit()
+
+    setup_postgres_connection_task = setup_postgres_connection()
+
+    @task
+    def setup_sheets_connection():
+        conn = Connection(
+            conn_id=SHEETS_CONNECTION_ID,
+            conn_type="google_cloud_platform",
+        )
+        conn_extra = {
+            "scope": "https://www.googleapis.com/auth/spreadsheets,https://www.googleapis.com/auth/cloud-platform",
+            "project": PROJECT_ID,
+            "keyfile_dict": "",  # Override to match your needs
+        }
+        conn_extra_json = json.dumps(conn_extra)
+        conn.set_extra(conn_extra_json)
+
+        session: Session = Session()
+        session.add(conn)
+        session.commit()
+
+    setup_sheets_connection_task = setup_sheets_connection()
```

Comment:
Do you think we could move repeated code into a generic function and just call it in tasks?